modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/ComponentBasedIdHelper.java

```diff
@@ -56,4 +61,50 @@ static String sourceElementNameFromSimpleValue(ComponentAst owner, ComponentPara
     return elementModel.getRawParameterValue(CONFIG_ATTRIBUTE_NAME);
   }
 
+  public static int computeHashFor(ComponentParameterAst componentParameterAst) {
+    return ParameterVisitorFunctions.computeHashFor(componentParameterAst);
+  }
+
+  private static class ParameterVisitorFunctions {
+
+    private static int computeHashFor(ComponentParameterAst parameter) {
+      return hash(new ParameterVisitorFunctions(parameter).hashBuilder.toString());
+    }
+
+    private StringBuilder hashBuilder = new StringBuilder();
+    private final Function<String, Void> leftFunction = this::hashForLeft;
+    private final Function<Object, Void> rightFunction = this::hashForRight;
+
+    private ParameterVisitorFunctions(ComponentParameterAst startingParameter) {
+      startingParameter.getValue().reduce(leftFunction, rightFunction);
+    }
+
+    private Void hashForLeft(String s) {
+      hashBuilder.append(s);
+      return null;
+    }
+
+    private Void hashForRight(Object o) {
+      if (o instanceof ComponentAst) {
+        final ComponentAst c = (ComponentAst) o;
+        c.getParameters().stream().sorted(comparing(p -> p.getModel().getName())).forEach(p -> {
+          hashBuilder.append(p.getModel().getName());
+          if (p.getModel().getType() instanceof ArrayType) {
+            hashForList((Collection<ComponentAst>) p.getValue().getRight());
```

Comment:
How about Map?

Reply:
Aren't maps ObjectTypes? It will call the same function again but with the value.

Reply:
You are right